### DISCUSSION TITLE
How do I properly use regex for header contains inbox rule

### DISCUSSION THREAD

**Discussion by mehmetkursataydin** [tutao/tutanota#4145] [Q&A] [answered]
I am using Firefox relay, so first of all I added relay email to 'not spam' rule.

Next thing I want to do is add inbox rule with **header contains**, the reason I choose it because most of the emails will be coming from Firefox relay, the only way to differentiate between email sources is to check headers I believe, because from section includes something like: ""support@indiegala.com [via Relay]"" <noreply@relay.firefox.com> If I select From/Sender as field option in inbox rules that only accepts a valid email which it wouldn't be suitable for my case.

So I tried:
Field:
Header contains

Value:
From: ""support@indiegala.com [via Relay]"" <noreply@relay.firefox.com>

Target folder:
my target folder

but this doesn't work, so I tried it with regex in value field like:

/From: ""support@indiegala.com [via Relay]""/

this also didn't work.

So what is the correct way to implement what I want to achieve?

Also attaching part of the headers of an example email from Firefox relay below.

![image](https://user-images.githubusercontent.com/36981992/168794780-62e115ff-a610-4d47-915a-23abb8c436be.png)

**Comment by charlag** [comment, marked as answer]
Hi, probably a lot of characters (incl. dots and square brackets) don't do what you expect. Please check https://regex101.com/ with Javascript syntax

**Reply by mehmetkursataydin** [reply]
I simply changed my regex pattern to /From: ""support/ just to test and the website you sent has a match but tutanota is still not catching it.

**Reply by mehmetkursataydin** [reply]
My bad, just figured out that I have a rule above this rule which was getting executed... Got it fixed, thanks.
